pkg/kubectl/util/deployment/deployment.go

```diff
@@ -68,6 +79,30 @@ func Revision(obj runtime.Object) (int64, error) {
 	return strconv.ParseInt(v, 10, 64)
 }
 
+// Revisions returns the revisions associated with the input object.
+func Revisions(obj runtime.Object) ([]int64, error) {
+	acc, err := meta.Accessor(obj)
+	if err != nil {
+		return nil, err
+	}
+
+	versions := []int64{}
+
+	if v, ok := acc.GetAnnotations()[RevisionAnnotation]; ok {
+		if versionNumber, err := strconv.ParseInt(v, 10, 64); err == nil {
+			versions = append(versions, versionNumber)
+		}
+	}
+	if v, ok := acc.GetAnnotations()[RevisionHistoryAnnotation]; ok {
```

Comment:
`RevisionHistoryAnnotation` doesn't matter here, it's just the previous revision number of this specific RS. `RevisionAnnotation` stores the revision number that matters for rollout history or rollback.

Reply:
example:
1. deploy revision 1. rs with "revision=1" exists.
2. deploy revision 2. rs with "revision=2" exists.
3. rollback to revision 1. rs with "revision=1" is updated to have "revision=3,historicalRevisions=1"
4. rollback to revision 2. rs with "revision=2" is updated to have "revision=4,historicalRevisions=2"

if we don't look at the historical revisions annotation, you can no longer roll back to revision 1 again, despite us having the data to be able to do so. is that what we want?

Reply:
> if we don't look at the historical revisions annotation, you can no longer roll back to revision 1 again, despite us having the data to be able to do so. is that what we want?

Yes this is what we want. In this example, `kubectl rollout history` only lists rv3 and rv4, and users can choose to roll back to rv3 or rv4 (rolling back to rv4 will be no-op).